core/common/src/geometry/ElementGeometry.ts

```diff
@@ -1093,6 +1112,12 @@ export namespace ElementGeometry {
       builder.addTransform(fbb, transformOffset);
     }
 
+    if (glyphs) {
+      builder.addRange(fbb, EGFBAccessors.TextStringRange.createTextStringRange(fbb, 0, 0, 20741, 2500));
```

Comment:
This hard-coded range can't be right...probably should ask Jeff Marker about this/whether it's needed since we can't compute it in typescript.

Reply:
Sorry, it was something I did for testing. Will correct it

Reply:
Looks like it has to be passed as a property as well.

Reply:
The C++ code populates the range together with glyphs, so I assume it's needed

Reply:
I still think it would be better to go with my initial suggestion and just to add an optional "glyph data" arg to ElementGeometry.Builder.appendTextString/ElementGeometry.fromTextString instead of adding things to TextString/TextStringProps that aren't generally useful and are going to be ignored by the json GeometryStreamBuilder now. Any opinion @pmconne ?

Reply:
Lacking some context, but I take it there is no desire/need to retrieve the glyph data from a persistent geometry stream, only to write it to a new geometry stream?

Reply:
Correct, this is strictly to preserve the V8 glyph layout when it uses options that are no longer supported bt itwin. Only the native code needs to look at it for display and it's lost on edit.

Reply:
The only reason I went with this was for unit tests. But I can create new tests in `core-common` rather than modifying the ones in `core-backend`

Reply:
You will still have a class/props for the glyph data in common just like TextString/TextStringProps that can be used by the backend. Not sure why you think it affects unit tests?

Reply:
Yes, I can create tests in `core-common`. The current code modifies existing unit tests in `core-backend`

Reply:
I don't understand why you think you can't continue to create the tests in core-backend. All the geometry stream tests are in core-backend...there is no reason this one needs to be in common, please explain???

Reply:
Right. `core-backend` does not import `flatbuffer` and `EGFBAccessors` which is internal to `core-common`. I needed them to read the flatbuffer.

Reply:
NOTE: The current "create GeometricElement3d from world coordinate text using a newly added font" test is using the json builder, which is not going to persist the glyph data (because the backend native fromJSON code implementation)...you will need to write a new text test for the flatbuffer builder.

Reply:
It doesn't need to, you are calling ElementGeometry.Builder/Iterator methods.

Reply:
It looks to me that the current "create GeometricElement3d from world coordinate text using a newly added font" test is calling `fromTextString` which builds the flatbuffer with the glyphs and passing it to `insertElement`. So it does persist the glyph data....?
But anyways. I can revert the changes in `GeometryStream.test.ts` since it doesn't have a way to read the flat buffer if the reading code is removed from `toTextString`. Then, I can add the tests in the `core-common` package

Reply:
> So it does persist the glyph data....?

Yes, the flatbuffer test should persist the data, the json test/builder does not...which is one of the reasons I don't think this should be added to TextStringProps.

Feel free to call me if this still isn't clear...